src/common/install.ts

```diff
@@ -0,0 +1,160 @@
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { print1, print2 } from "../drivers/stdio";
+import { isa } from "../types";
+import { compact } from "lodash";
+import { spawn } from "node:child_process";
+import os from "node:os";
+import { sys } from "typescript";
+import which from "which";
+
+export const SupportedPlatforms = ["darwin"] as const;
+export type SupportedPlatform = (typeof SupportedPlatforms)[number];
+
+export const AwsItems = ["aws"] as const;
+export type AwsItem = (typeof AwsItems)[number];
+
+export type InstallMetadata = {
+  label: string;
+  commands: Record<SupportedPlatform, Readonly<string[]>>;
+};
+
+export const AwsInstall: Readonly<Record<AwsItem, InstallMetadata>> = {
+  aws: {
+    label: "AWS CLI v2",
+    commands: {
+      darwin: [
+        'curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"',
+        "sudo installer -pkg AWSCLIV2.pkg -target /",
+        'rm "AWSCLIV2.pkg"',
+      ],
+    },
+  },
+};
+
+const printToInstall = <
+  T extends string,
+  U extends Readonly<Record<T, InstallMetadata>>,
+>(
+  toInstall: readonly T[],
+  installMetadata: U
+) => {
+  print2("The following items must be installed on your system to continue:");
+  for (const item of toInstall) {
+    print2(`  - ${installMetadata[item].label} (${item})`);
+  }
+  print2("");
+};
+
+const queryInteractive = async () => {
+  const inquirer = (await import("inquirer")).default;
+  const { isGuided } = await inquirer.prompt([
+    {
+      type: "confirm",
+      name: "isGuided",
+      message:
+        "Do you want P0 to install these for you (sudo access required)?",
+    },
+  ]);
+  print2("");
+  return isGuided;
+};
+
+const requiredInstalls = async <T extends string>(installItems: readonly T[]) =>
+  compact(
+    await Promise.all(
+      installItems.map(async (item) =>
+        (await which(item, { nothrow: true })) === null ? item : undefined
+      )
+    )
+  );
+
+const printInstallCommands = <
+  T extends string,
+  U extends Readonly<Record<T, InstallMetadata>>,
+>(
+  platform: SupportedPlatform,
+  item: T,
+  installData: U
+) => {
+  const { label, commands } = installData[item];
+  print2(`To install ${label}, run the following commands:\n`);
+  for (const command of commands[platform]) {
+    print1(`  ${command}`);
+  }
+  print1(""); // Newline is useful for reading command output in a script, so send to /fd/1
```

Comment:
Why print1 and not print2 here?

Reply:
This is copied from the [original implementation](https://github.com/p0-security/p0cli/blob/2ae3841b9ae920cbee4f0dd895964270af85ab31/src/plugins/aws/ssm/install.ts) that I basically just moved here. Based on the comment, I presume the intention was to send to stdout so that a harness script or something could handle installation of the tool instead of the P0 CLI itself, but I can only presume alas.